attributes/default.rb

```diff
@@ -7,6 +7,8 @@
 # Init system to use
 default['mesos']['init']    = case node['platform']
                               when 'debian' then 'sysvinit_debian'
+                              when 'centos', 'amazon', 'scientific'
+                                node['platform_version'].to_i >= 7 ? 'systemd' : 'upstart'
```

Comment:
CentOS/RHEL do not use upstart. CentOS/RHEL versions < 7 uses sysvinit system, so this would fail on CentOS versions < 7.